go-controller/pkg/libovsdb/util/acl.go

```diff
@@ -214,3 +215,36 @@ const (
 	// emptyProtocol is used to create ACL for gressPolicy that doesn't have port policies hence no protocols
 	EmptyProtocol = "None"
 )
+
+// convertK8sProtocolToOVNProtocol returns the OVN syntax-specific protocol value for a v1.Protocol K8s type
+func convertK8sProtocolToOVNProtocol(proto v1.Protocol) string {
+	var protocol string
+	switch proto {
+	case v1.ProtocolTCP:
+		protocol = "tcp"
+	case v1.ProtocolSCTP:
+		protocol = "sctp"
+	case v1.ProtocolUDP:
+		protocol = "udp"
+	}
+	return protocol
+}
+
+// NetworkPolicyPort is an internal representation of
+// knet.NetworkPolicyPort and anpapi.AdminNetworkPolicyPort
+// in a simpler representation format for the caches
+type NetworkPolicyPort struct {
+	Protocol string
+	Port     int32 // will store startPort if its a range
+	EndPort  int32
+}
+
+// GetNetworkPolicyPort returns an internal NetworkPolicyPort struct
+// It also sets the provided protocol field
+func GetNetworkPolicyPort(proto v1.Protocol) *NetworkPolicyPort {
```

Comment:
Maybe `GetNetworkPolicyPort(proto v1.Protocol, port, endport *int32)` ?

or even a more helpful `GetNetworkPolicyPort(proto v1.Protocol, port, startport, endport *int32)` so you can resolve whether it is a range or not within the utility, up to you

Reply:
ack yep that sounds like a better util.. changing..

Reply:
there is a still hiccup here because netpol's port is of type `Port *intstr.IntOrString` and ANP's is all int32, its icky to send them in here without conversion so there is only so much we can commonize here...

Reply:
I'll do a push with what I managed and happy to take an suggestions you might have..